src/app/pages/admin/blog/page.tsx

```diff
@@ -1,11 +1,137 @@
-import React from "react";
+"use client";
+import { useState, useEffect } from "react";
 import DashBoardLayout from "@/components/layout/layout";
+import { DataTable } from "@/components/DashBoard/Table";
+import { ColumnDef } from "@tanstack/react-table";
+import { CiEdit } from "react-icons/ci";
+import { MdDeleteForever } from "react-icons/md";
+import { Button } from "@/components/ui/button";
+import { ListBlog } from "@/types/Types";
+import Image from "next/image";
+import DialogConfirm from "@/components/Confirm/confirm";
+import {
+  useGetListThumbnailQuery,
+  useGetListBlogQuery,
+  useDeleteBlogMutation,
+} from "@/lib/api/services/AdminBlog";
+import { IoAddCircleOutline } from "react-icons/io5";
 
 const Blog = () => {
+  const [dialogOpen, setDialogOpen] = useState(false);
+
+  const [dialogConfirm, setDialogConfirm] = useState(false);
+  const [storeId, setStoreId] = useState<number | null>(null);
+
+  const { data: thumbnail, refetch: refetchThumbnail } =
+    useGetListThumbnailQuery();
+
+  const { data: listBlog, refetch: refetchBlog } = useGetListBlogQuery();
+  const [deleteBlog] = useDeleteBlogMutation();
+
+  useEffect(() => {
+    refetchThumbnail();
+
+    refetchBlog();
+  }, [refetchThumbnail, refetchBlog]);
```

Comment:
no need to refetch, do in invalidation tag the same, so it will change automatically.